From da8f70efb66866f929594650943b3425d2e5e858 Mon Sep 17 00:00:00 2001
From: Siarhei Siamashka <siarhei.siamashka@gmail.com>
Date: Mon, 7 Dec 2015 11:37:37 +0200
Subject: [PATCH 01/27] Fix compilation problems with more modern version of
 GCC

Now the kernel can be compiled at least with GCC 4.7

Signed-off-by: Siarhei Siamashka <siarhei.siamashka@gmail.com>
---
 arch/arm/mach-sunxi/power/brom/Makefile | 2 +-
 arch/arm/mm/dma-mapping.c               | 2 +-
 drivers/gpu/ion/ion_cma_heap.c          | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm/mach-sunxi/power/brom/Makefile b/arch/arm/mach-sunxi/power/brom/Makefile
index 5bce9ee..b7b36d8 100755
--- a/arch/arm/mach-sunxi/power/brom/Makefile
+++ b/arch/arm/mach-sunxi/power/brom/Makefile
@@ -5,7 +5,7 @@ targets := resumes.elf
 
 #use "-Os" flags.
 #Don't use "-O2" flags.
-KBUILD_CFLAGS := -g -c -nostdlib -march=armv7-a -marm -fno-unwind-tables -fno-asynchronous-unwind-tables -mlittle-endian -O2 --min_array_alignment=4 --no_unaligned_access
+KBUILD_CFLAGS := -g -c -nostdlib -march=armv7-a -marm -fno-unwind-tables -fno-asynchronous-unwind-tables -mlittle-endian -O2
 
 #Include the cur dir.
 KBUILD_CPPFLAGS += -I.
diff --git a/arch/arm/mm/dma-mapping.c b/arch/arm/mm/dma-mapping.c
index 5e2530c..ee9a374 100755
--- a/arch/arm/mm/dma-mapping.c
+++ b/arch/arm/mm/dma-mapping.c
@@ -576,7 +576,7 @@ static void __free_from_contiguous(struct device *dev, struct page *page,
 	dma_release_from_contiguous(dev, page, size >> PAGE_SHIFT);
 }
 
-inline pgprot_t __get_dma_pgprot(struct dma_attrs *attrs, pgprot_t prot)
+pgprot_t __get_dma_pgprot(struct dma_attrs *attrs, pgprot_t prot)
 {
 	prot = dma_get_attr(DMA_ATTR_WRITE_COMBINE, attrs) ?
 			    pgprot_writecombine(prot) :
diff --git a/drivers/gpu/ion/ion_cma_heap.c b/drivers/gpu/ion/ion_cma_heap.c
index 86612c0..2e95a9d 100755
--- a/drivers/gpu/ion/ion_cma_heap.c
+++ b/drivers/gpu/ion/ion_cma_heap.c
@@ -48,7 +48,7 @@ struct ion_cma_buffer_info {
 struct page *dma_alloc_from_contiguous(struct device *dev,
 				int count, unsigned int align);
 void __dma_clear_buffer(struct page *page, size_t size);
-inline pgprot_t __get_dma_pgprot(struct dma_attrs *attrs, pgprot_t prot);
+pgprot_t __get_dma_pgprot(struct dma_attrs *attrs, pgprot_t prot);
 void *__dma_alloc_remap(struct page *page, size_t size, gfp_t gfp,
 				pgprot_t prot, const void *caller);
 void __dma_remap(struct page *page, size_t size, pgprot_t prot);
-- 
1.9.1

